File: solutions/Python/13.RomanToInteger.py
class Solution:
    def romanToInt(self, s: str) -> int:
        result = 0
        length = len(s)

        for i in range(length):
            if i < length - 1 and (
                (s[i] == "I" and s[i + 1] in ["V", "X"])
                or (s[i] == "X" and s[i + 1] in ["L", "C"])
                or (s[i] == "C" and s[i + 1] in ["D", "M"])
            ):
                result -= self.roman_switch(s[i])
            else:
                result += self.roman_switch(s[i])

        return result

    def roman_switch(self, char: str) -> int:
        switch = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
        return switch.get(char, 0)  # 0 if the character is not in the dictionary

File: solutions/TypeScript/13.RomanToInteger.ts
function romanToInt(s: string): number {
  let result: number = 0;

  for (let i: number = 0; i < s.length; i++) {
    if (
      (s[i] === "I" && ["V", "X"].includes(s[i + 1])) ||
      (s[i] === "X" && ["L", "C"].includes(s[i + 1])) ||
      (s[i] === "C" && ["D", "M"].includes(s[i + 1]))
    ) {
      result -= getRomanValue(s[i]);
    } else {
      result += getRomanValue(s[i]);
    }
  }

  return result;
}

function getRomanValue(char) {
  switch (char) {
    case "I":
      return 1;
    case "V":
      return 5;
    case "X":
      return 10;
    case "L":
      return 50;
    case "C":
      return 100;
    case "D":
      return 500;
    case "M":
      return 1000;
    default:
      return 0;
  }
}
